### PR TITLE
feat(FX-3526): enable the Sort By option for the Artworks pill when feature flag is enabled

### PR DESCRIPTION
**Pull request by dimatretyak**
The type of this PR is: **Feature**

This PR resolves [FX-3526]

⚠️ Changes behind a feature flag

### Description
* As a user
* When I open the improved search screen
* And when the feature flag is enabled
* And when I press the "sort & filter" button on the artworks pill
* I see the Sort By option and I can specify a sort filter

### Acceptance criteria
* Changes are available under the feature flag
* if feature flag is **enabled**, display **Sort & Filter** label
* if feature flag is **disabled**, display **Filter** label
* Changes should be behind a feature flag

### Useful links
* [Slack](https://artsy.slack.com/archives/C9SATFLUU/p1635860066395500)
 
#### Before
https://user-images.githubusercontent.com/3513494/139913321-74d8425e-f388-4bb9-b851-b57b5a6ca84d.mp4

#### After
https://user-images.githubusercontent.com/3513494/139913352-c0fc8ab1-5476-440d-8af7-f456d7b4fd3a.mp4

### PR Checklist (tick all before merging)

- [x] I have included screenshots or videos to illustrate my changes, or I have not changed anything that impacts the UI.
- [x] I have tested my changes on **iOS** and **Android**.
- [x] I have added tests/stories for my changes, or my changes don't require testing/stories, or I have included a link to a separate Jira ticket covering the tests.
- [x] I have added a feature flag, or my changes don't require a feature flag. ([How do I add one?](https://github.com/artsy/eigen/blob/master/docs/developing_a_feature.md))
- [x] I have documented any follow-up work that this PR will require, or it does not require any.
- [x] I have added an app state migration, or my changes do not require one. ([What are migrations?](https://github.com/artsy/eigen/blob/master/docs/adding_state_migrations.md))
- [x] I have added a changelog entry below or my changes do not require one.

### To the reviewers 👀

- [ ] I would like at least one of the reviewers to run this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

- Enable the Sort By option for the Artworks pill when feature flag is enabled - dimatretyak

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

[FX-3526]: https://artsyproduct.atlassian.net/browse/FX-3526?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
### This PR contains the following changes:

- Cross-platform user-facing changes (Enable the Sort By option for the Artworks pill when feature flag is enabled - dimatretyak)
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 2507942541264102f7f3d572264ddb8547bd9e2a
</p>

File: src/lib/Components/ArtworkFilter/ArtworkFilterOptionsScreen.tsx [hunk 1]
```diff
@@ -233,7 +236,16 @@ export const getStaticFilterOptionsByMode = (mode: FilterModalMode) => {
         filterOptionToDisplayConfigMap.organizations,
       ]
 
+    // TODO: Remove the entire case block when AREnableSortFilterForArtworksPill flag is released
```
Comment (gkartalis) [inline src/lib/Components/ArtworkFilter/ArtworkFilterOptionsScreen.tsx:239]: useful comment 🤩

File: src/lib/Scenes/Search2/SearchArtworksGrid.tsx [hunk 2]
```diff
@@ -86,7 +88,7 @@ const SearchArtworksGrid: React.FC<SearchArtworksGridProps> = ({ viewer, relay,
         mode={FilterModalMode.Search}
       />
       <ArtworksFilterHeader
-        title="Filter"
+        title={enableSortFilter ? "Sort & Filter" : "Filter"}
```
Comment (gkartalis) [inline src/lib/Scenes/Search2/SearchArtworksGrid.tsx:91]: **suggestion:**  when we pass a title the `ArtworksFilterHeader` component renders the title, when it is undefined it renders the standard `"Sort & Filter"` title. What do you think of maybe changing this like:
```suggestion
        {...!enableSortFilter && { title: "Filter" }}
```

File: src/lib/store/config/features.ts [hunk 3]
```diff
@@ -186,6 +186,12 @@ export const features = defineFeatures({
     description: "Enable Split.io A/B testing",
     showInAdminMenu: true,
   },
+  AREnableSortFilterForArtworksPill: {
+    readyForRelease: false,
```
Comment (gkartalis) [inline src/lib/store/config/features.ts:190]: **question:** Should we mark it as readyForRelease and do a PR in echo as well for that with the flag turned to false for now to get it out of the way?

**Review by gkartalis — APPROVED** [review]

File: src/lib/store/config/features.ts (hunk 3 above)
Comment (dimatretyak) [reply src/lib/store/config/features.ts:190]: Good point. Yesterday I forgot to mark it as ready for release. PR for Echo was merged yesterday (artsy/echo#163)

**Pull request merged**